modules/generic/testcontainers/generic/providers/sql_connector.py

```diff
@@ -0,0 +1,48 @@
+# This module provides a wait strategy for SQL database connectivity testing using SQLAlchemy.
+# It includes handling for transient exceptions and connection retries.
+
+import logging
+
+from testcontainers.core.waiting_utils import WaitStrategy, WaitStrategyTarget
+
+logger = logging.getLogger(__name__)
+
+ADDITIONAL_TRANSIENT_ERRORS = []
+try:
+    from sqlalchemy.exc import DBAPIError
```

Comment:
Should we consider moving this check into the initializer below?  

If `sqlalchemy` is unavailable, at best a caller could construct an instance of the below class, and it'll raise when `wait_until_ready` is eventually called.  Unless I'm misunderstanding - there doesn't seem to be any value to constructing an instance where `ADDITIONAL_TRANSIENT_ERRORS` is empty?

Reply:
ideally one of the benefits of moving this to its own corner (so it can sit and think about what it did) is that sqlalchemy is no longer conditionally available (it either is in the case of the extra being installed, or no)